price-service/src/servicePoll.ts

```diff
@@ -0,0 +1,15 @@
+import { PriceMonitor } from "./priceMonitor";
+
+const { HermesClient } = require('@pythnetwork/hermes-client');
+const fs = require('fs');
+
+const config = JSON.parse(fs.readFileSync('config.json', 'utf8'));
+const priceMonitor = new PriceMonitor(config);
+const connection = new HermesClient(config.pyth_url);
+
+async function poll() {
+    const priceUpdates = await connection.getLatestPriceUpdates(config.priceIds);
```

Comment:
what if this has an error?

Reply:
For now i use pm2 so crash and restart the app, Since it has never failed and they do not present any error handling in their examples i figured this might be the most stable way of doing it. Could catch and exit more gracefully perhaps?

Reply:
for graceful error catching something like this can be implemented

https://github.com/icon-community/xp-icon-community-backend/blob/master/db-manager/blockchain-scraper.js#L133